neuralpp/test/quick_tests/graphical_model/representation/factor/symbolic_factor_test.py

```diff
@@ -21,7 +21,9 @@ def test_sympy_condition():
 
     symbolic = SymbolicFactor([x, y], expression1)
     conditioned = symbolic.condition({x: 1})
-    assert conditioned.expression == SymPyVariable(y_symbol, int)
+
+    expected = SymPyVariable(y_symbol, int)
+    assert conditioned.expression == expected
```

Comment:
This is probably not really checking if the expressions are equal, it is just making a new `Expression` that is an equality. We need to use `syntactic_eq` for all assertions of `Expression` comparisons.